packages/flutter/lib/src/material/expansion_panel.dart

```diff
@@ -446,6 +470,24 @@ class _ExpansionPanelListState extends State<ExpansionPanelList> {
         context,
         _isChildExpanded(index),
       );
+      final Widget expansionIndicator = child.expansionIndicator != null
+          ? InkResponse(
+        onTap: !child.canTapOnHeader
+            ? () => _handlePressed(_isChildExpanded(index), index)
+            : null,
+        child: child
+            .expansionIndicator(
+          context,
+          _isChildExpanded(index),
+        ),
+      )
+          : ExpandIcon(
+        isExpanded: _isChildExpanded(index),
+        padding: const EdgeInsets.all(16.0),
+        onPressed: !child.canTapOnHeader
```

Comment:
Do not use arrow functions when not explicitly returning a value, since it could be confusing to first-time readers of this code. Instead, use regular function expressions.

Reply:
Done; This part of the code was copied from  original;  Anyway, I think what you said is right and  reasonable